File: test-case/common/intersect/Intersect.scala
import soda.scala.unittest._

import scala.collection.mutable
import scala.reflect.runtime.universe.typeOf

class Solution {}
object Solution {
  def intersection(nums1: Array[Int], nums2: Array[Int]): Array[Int] = {
    if (nums1.size > nums2.size) {
      return intersection(nums2, nums1)
    }

    val mset = mutable.Set[Int]()
    val res = mutable.Set[Int]()
    for (n <- nums1) {
      mset += n
    }
    for (b <- nums2) {
      if (mset.contains(b)) {
        res += b
      }
    }
    res.toArray
  }
}

class Intersect extends (String => String) {
  override def apply(text: String): String = {
    val work = TestWork.forObject(typeOf[Solution], "intersection")
    // val work = TestWork.forStruct(...)
    work.setValidator(Validators.forArray[Int](false))
    work.compareSerial = true
    work.run(text)
  }
}

object Intersect extends App {
  println(new Intersect()(Utils.fromStdin()))
}

File: test-case/common/intersect/intersect.py
#!/usr/bin/env python3
from collections import Counter, defaultdict, deque
import heapq
import json
import logging
import math
import random
import time
from typing import *
import sys

from soda.leetcode.bitree import *
from soda.leetcode.graph import *
from soda.leetcode.linklist import *
from soda.unittest.util import init_logging

logger = logging.getLogger(__name__)

# step [1]: implement class Solution
# class Solution: pass
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        if len(nums1) > len(nums2):
            return self.intersection(nums2, nums1)
        mset = set()
        res = set()
        for n in nums1:
            mset.add(n)
        for b in nums2:
            if b in mset:
                res.add(b)
        return list(res)

if __name__ == '__main__':
    init_logging()
    from soda.unittest import Validators
    from soda.unittest.work import TestWork

    # step [2]: setup function
    # Attention! FUNCTION must use type hint, including arguments and return type
    work = TestWork(Solution().intersection)
    # OR use struct tester
    # work = TestWork.forStruct(CLASS)

    # step [3]: setup other options
    work.validator = Validators.forList(int, False)
    work.compareSerial = True
    work.run()

